[Tejas] iP

tejas-1405 opened the pull request:
> One of the chatbots of all time

# Tars Chatbot

This is a **chatbot** application that allows you to _manage tasks_.

  ## Installation

1. Clone the repo
2. Compile all java files

   `public static void main(String[] args) throws DukeException {
        new Duke().run();
    }`

It's that easy :+1:


## Adding deadlines

- Create a Todo task using the follow format

Example: `todo taskname`


```
Created todo task: taskname
```


## Features

- Task Management
- Task loading and saving

- This site was built using [GitHub Pages](https://pages.github.com/).

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

tejas-1405 committed:
Completed Level-0

tejas-1405 committed:
Level 1 done

tejas-1405 committed:
Added Level-2

tejas-1405 committed:
Added Level 3

tejas-1405 committed:
Added Level 4

tejas-1405 committed:
A-textUiTesting added

tejas-1405 committed:
Level-5

tejas-1405 committed:
Added Level-6

tejas-1405 committed:
Add Level-7

tejas-1405 committed:
Add Level-8

tejas-1405 committed:
Merge branch 'branch-Level-8'

tejas-1405 committed:
Add more OOP design

tejas-1405 committed:
Add package

tejas-1405 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

tejas-1405 committed:
Add Gradle

tejas-1405 committed:
Add Testcases

ReganChoy commented on src/main/java/duke/InputHandler.java:
```diff
@@ -0,0 +1,27 @@
+
+
+
+
```
Would it be better to remove this big empty space?

ReganChoy commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,72 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
```
Great use of protected access modifier! Naming of variables are also correct so far!

ReganChoy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,221 @@
+    }
+
+    private static List<Task> retrieveData(File file) throws IOException {
+        List<Task> loadedList = new ArrayList<>();
```
Methods that do something all start with a verb! Good job

ReganChoy commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,221 @@
+        return loadedList;
+    }
+
+    private static void write(String filePath, List<Task> tasks) throws IOException {
```
Excellent use of private methods where not all methods are public.

ReganChoy commented on src/main/java/duke/InputHandler.java:
```diff
@@ -0,0 +1,27 @@
+import java.time.LocalDate;
+public class InputHandler {
+    public LocalDate formatDeadline(String[] data) throws ArrayIndexOutOfBoundsException{
+
```
All functions and classes are named clearly

tejas-1405 committed:
Add JavaDocs

tejas-1405 committed:
Add coding standard and taskList

tejas-1405 committed:
Fixed conflict

tejas-1405 committed:
Update README.md

tejas-1405 committed:
Fix bugs

tejas-1405 committed:
Merge branch 'master' of https://github.com/tejas-1405/ip

0liverkong commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,105 @@
+    public static Task parser(String taskString) {
+        String str = taskString.substring(1,2);
+
+        switch (str) {
```
the indentation for the cases need to be at the same level as the switch

0liverkong commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,115 @@
+    public void retrieveData(File file) throws IOException {
+        List<Task> loadedList = new ArrayList<>();
+
+        try (Scanner fileScanner = new Scanner(file)) {
```
the try statement must have a catch or a catch + finally after

0liverkong commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+            return "[D]" + super.printWithStatus() + " (by: " +
+                    date.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+        }
+        else {
```
the else should follow right after the }

0liverkong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,201 @@
+                tlist.listTasks();
+            }
+
+            else if (comd.startsWith("todo ")) {
```
same for all these else if clauses here, they should follow in the same line as the closing bracket of the previous clause.

tejas-1405 committed:
Added GUI

tejas-1405 committed:
Bug fix

tejas-1405 committed:
Add Storage

tejas-1405 committed:
Add Storage file

tejas-1405 committed:
Integrate GUI

tejas-1405 committed:
GUI has been integrated along with the chat logic

Users can now interact with the GUI rather than the CLI

This improves the user experience

Let's make the chatbot greet the user with an opening line as right now the user is the first one to send the message

tejas-1405 committed:
Add Assertions

tejas-1405 committed:
Improve Code Quality

tejas-1405 committed:
Merge pull request #2 from tejas-1405/branch-A-Assertions

Add Assertions

tejas-1405 committed:
Merge branch 'master' into branch-A-CodeQuality

tejas-1405 committed:
Merge pull request #3 from tejas-1405/branch-A-CodeQuality

Improve Code Quality

tejas-1405 committed:
Add greeting

tejas-1405 committed:
Add Help extension

tejas-1405 committed:
Improve GUI

tejas-1405 committed:
Update README.md

tejas-1405 committed:
Update README.md

tejas-1405 committed:
Improve GUI

tejas-1405 committed:
Merge branch 'master' of https://github.com/tejas-1405/ip

tejas-1405 committed:
Change package names

tejas-1405 committed:
Add screenshot

tejas-1405 committed:
Add UG

tejas-1405 committed:
Create JAR file

tejas-1405 committed:
Finish UG

tejas-1405 committed:
Update README.md

tejas-1405 committed:
Fix minor issue
